GHzDACs/jump_table.py

```diff
@@ -0,0 +1,480 @@
+# Author: Daniel Sank
+# Created: July 2013
+
+import numpy as np
+from util import littleEndian
+
+IDLE_NUM_BITS = 15
+IDLE_MIN_CYCLES = 0
+IDLE_MAX_CYCLES = (2 ** IDLE_NUM_BITS) - 1
+
+
+class JumpEntry(object):
+    """ A single entry in the jump table.
+
+    """
+
+    def __init__(self, from_addr, to_addr, operation):
+        """ Create a single jump table entry.
+
+        :param int from_addr: the from address
+        :param int to_addr: the to address (should be 0 if not used)
+        :param Operation operation: the operation
+        """
+        if abs(from_addr - to_addr) - 1 == 0:
+            raise ValueError(
+                "from_addr: {} and to_addr: {} are too close".format(
+                    from_addr, to_addr)
+            )
+
+        self.from_addr = from_addr
+        self.to_addr = to_addr
+        self.operation = operation
+
+    def __str__(self):
+        f_str = "from_addr: %d" % self.from_addr
+        t_str = "to_addr: %d" % self.to_addr
+        op_str = str(self.operation)
+        return '\n'.join([f_str, t_str, op_str])
+
+    def as_bytes(self):
```

Comment:
Man, I'm such a derp. I didn't even notice that the return type is already documented because it's in pycharm format and my meat brain still can't read those darn things.